controllers/depresolver/depresolver_config.go

```diff
@@ -70,13 +74,23 @@ func (dr *DependencyResolver) ResolveOperatorConfig() (*Config, error) {
 		dr.config.Infoblox.HTTPRequestTimeout, _ = env.GetEnvAsIntOrFallback(InfobloxHTTPRequestTimeoutKey, 20)
 		dr.config.Override.FakeDNSEnabled = env.GetEnvAsBoolOrFallback(OverrideWithFakeDNSKey, false)
 		dr.config.Override.FakeInfobloxEnabled = env.GetEnvAsBoolOrFallback(OverrideFakeInfobloxKey, false)
+		dr.config.Log.Level, _ = zerolog.ParseLevel(strings.ToLower(env.GetEnvAsStringOrFallback(LogLevelKey, zerolog.InfoLevel.String())))
+		dr.config.Log.OutputFormat = parseLogOutputFormat(strings.ToLower(env.GetEnvAsStringOrFallback(LogOutputFormatKey, "mono")))
 		dr.errorConfig = dr.validateConfig(dr.config)
 		dr.config.EdgeDNSType = getEdgeDNSType(dr.config)
 	})
 	return dr.config, dr.errorConfig
 }
 
 func (dr *DependencyResolver) validateConfig(config *Config) (err error) {
+	if config.Log.Level == zerolog.NoLevel {
+		config.Log.Level = zerolog.InfoLevel
+		return fmt.Errorf("invalid %s, allowed values ['','trace','debug','info','warn','error','fatal','panic']", LogLevelKey)
```

Comment:
Customer makes typo and sets level to `debg` instead of `debug`. Rather than keeping logger in `info` we should inform him about typo and exit k8gb ASAP.

----

The topic is larger. If parser doesn't recognise level, then `NoLevel` returned.  `NoLevel` value disable logger completely. Similarly `Unrecognised` for unrecognised format. 

But fundamental question comes now: How to log error, when logger (or config) was not recognised. The natural solution is to always provide default values i.e. print error with  `simple + info + no_color` logger. I was deciding whether to handle `NoLevel / Unrecognised format` in logger factory, or handle in depresolver. 

Not sure now, maybe will move responsibility out depresolver and logger factory must take care.

Reply:
@kuritka, I see what you mean. If we throw a well-formatted error in that case and then shut down k8gb controller - that looks good.

Reply:
@somaritane We return formatted error message but with `NoLevel / Unrecognised`. Now it is on logger factory to create default logger with such states. 

pseudocode :

```go
config, err := depresolver.GetConfig()
l :=  NewLoggerFactory(config).Get() // this thing builds logger based on config.Log
if err!=nil {
 // we have default logger instance here ([json,info])
 l.Err().Msg("configuration error")   // writes configuration error  invalid  LOG_LEVEL, allowed values ['','trace',...]
}
```

Reply:
Thanks for clarification @kuritka !